Src/PerformanceCollector/Shared/Implementation/WebAppPerformanceCollector/CacheHelper.cs

```diff
@@ -0,0 +1,128 @@
+        /// </summary>
+        private CacheHelper()
+        {
+        }
+
+        /// <summary>
+        /// Gets the only instance of CacheHelper.
+        /// </summary>
+        public static CacheHelper Instance
+        {
+            get
+            {
+                return CacheHelperInstance;
+            }
+        }
+
+        /// <summary>
+        /// Search for the value of a given performance counter in a JSON.
+        /// </summary>
+        /// <param name="performanceCounterName"> The name of the performance counter.</param>
+        /// <param name="json"> String containing the JSON.</param>
+        /// <returns> Value of the performance counter.</returns>
+        public int PerformanceCounterValue(string performanceCounterName, string json)
+        {
+            if (json.IndexOf(performanceCounterName, StringComparison.OrdinalIgnoreCase) == -1)
+            {
+                throw new System.ArgumentException("Counter was not found.", performanceCounterName);
+            }
+
+            string jsonSubstring = json.Substring(json.IndexOf(performanceCounterName, StringComparison.OrdinalIgnoreCase), json.Length - json.IndexOf(performanceCounterName, StringComparison.OrdinalIgnoreCase));
+
+            int startingIndex = jsonSubstring.IndexOf(" ", StringComparison.Ordinal) + 1;
+            int value;
+            string valueString = string.Empty;
+
+            while (char.IsDigit(jsonSubstring[startingIndex]))
+            {
+                valueString += jsonSubstring[startingIndex];
+                startingIndex++;
+            }
+
+            if (!int.TryParse(valueString, out value))
+            {
+                throw new System.InvalidCastException("The value of the counter cannot be converted to integer type.");
+            }
+
+            return value;
+        }
+
+        /// <summary>
+        /// Checks if a key is in the cache and if not
+        /// Retrieves raw counter data from Environment Variables
+        /// Cleans raw JSON for only requested counter
+        /// Creates value for caching.
+        /// </summary>
+        /// <param name="name">Cache key and name of the counter to be selected from JSON.</param>
+        /// <param name="environmentVariable">Identifier of the environment variable.</param>
+        /// <returns>Value from cache.</returns>
+        public int GetCounterValue(string name, AzureWebApEnvironmentVariables environmentVariable)
+        {
+            if (!CacheHelper.Instance.IsInCache(name))
+            {
+                PerformanceCounterImplementation client = new PerformanceCounterImplementation();
+                string uncachedJson = client.GetAzureWebAppEnvironmentVariables(environmentVariable);
+
+                if (uncachedJson == null)
+                {
+                    return 0;
+                }
+
+                CacheHelper.Instance.SaveToCache(name, uncachedJson, DateTimeOffset.Now.AddSeconds(5.0));
```

Comment:
@kartang this is the reason that counters got stall for 5 seconds. You need to change this interval